restdocs-api-spec-jsonschema/src/test/kotlin/com/epages/restdocs/apispec/jsonschema/JsonSchemaFromFieldDescriptorsGeneratorTest.kt

```diff
@@ -289,6 +312,51 @@ class JsonSchemaFromFieldDescriptorsGeneratorTest {
         thenSchemaIsValid()
     }
 
+    @Test
+    fun should_specify_accurate_items_type_in_array_when_descriptor_contains_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithArrayOfSingleType()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(StringSchema::class.java)
```

Comment:
Relevant part of JSON schema spec:
https://json-schema.org/draft/2020-12/json-schema-core.html#items